src/vos/vos_query.c

```diff
@@ -393,6 +430,7 @@ vos_obj_query_key(daos_handle_t coh, daos_unit_oid_t oid, uint32_t flags,
 
 		dkey_punch = query.qt_punch;
 		dkey_epr = query.qt_epr;
+		vos_ts_set_save(query.qt_ts_set, &akey_save);
```

Comment:
Would you please to comment why need to save & restore the ts_set? Thanks!

Reply:
will do